### DISCUSSION TITLE
no more parameter "exclude=..." in ui.run (v1.3 and later)

### DISCUSSION THREAD

**Discussion by Derf000** [zauberzeug/nicegui#1182] [Q&A] [answered]
### Question

I noticed in v1.3 (and later) that the 'exclude=' parameter in ui.run() has disappeared. 

This seems to be linked to Feature/dependencies #658 , but I haven't fully 'understood' why.

Should this parameter be replaced by something else, or has it become totally obsolete?

**Comment by falkoschindler** [comment, marked as answer]
Hi @Derf000,
That's right, the `exclude` parameter has gone with version 1.3.
PR #658 changed the way dependencies are served: Now we only send components and libraries to the client that are actually needed by the elements on the page. So you don't need to explicitly exclude unused elements in `ui.run`. This saves a lot of bandwidth and reduces load times while preserving full flexibility. If your UI needs AG Grid in a very rare case, only then it will load the dependency.
Maybe we should have kept the parameter, showing a warning about it being ineffective since 1.3, instead of ["silently"](https://github.com/zauberzeug/nicegui/releases#:~:text=The%20exclude%20parameter%20of%20ui.run%20is%20gone.) dropping it.

**Reply by Derf000** [reply]
No problem for me, I was just trying to understand why this parameter was removed.
Thanks for all your work with nicegui... 👍
